mlflow/java/client/src/main/java/org/mlflow/tracking/MlflowClient.java

```diff
@@ -119,24 +119,58 @@ public RunInfo createRun(CreateRun request) {
   public List<RunInfo> listRunInfos(String experimentId) {
     List<String> experimentIds = new ArrayList<>();
     experimentIds.add(experimentId);
-    return searchRuns(experimentIds, null).stream().map(Run::getInfo)
-      .collect(Collectors.toList());
+    return searchRuns(experimentIds, null);
   }
 
   /**
-   * Return runs from provided list of experiments that satisfy the search query.
-   *
+   * Return RunInfos from provided list of experiments that satisfy the search query.
+   * @deprecated As of 1.1.0 - please use {@link #searchRuns(List, String, ViewType, int)} or
+   *                    similar that returns a page of Run results.
+   * 
    * @param experimentIds List of experiment IDs.
    * @param searchFilter SQL compatible search query string. Format of this query string is
```

Comment:
nit: Could we actually document everywhere which fields are nullable and what it means when it's null?

Reply:
The only parameter that would make sense to be null is `pageToken`. Even then a blank string will do the trick. When the null checks were added to the searchRunsV1 API, they basically let protobuf load in the default value by not specifying it. The default value is either specified in `service.proto` or it'll use the default value for the data type. 

With respect to this, should we throw `NullPointerException`s when null is passed in? Or should we just default to protobuf defaults? @andrewmchen @mparkhe